tern/analyze/default/dockerfile/run.py

```diff
@@ -68,16 +66,11 @@ def get_dockerfile_packages():
 def analyze_full_image(full_image, redo, driver, extension):
     """If we are able to load a full image after a build, we can run an
     analysis on it"""
-    # set up for analysis
-    crun.setup(full_image)
-    # analyze image
-    cimage.analyze(full_image, redo, driver, extension)
-    # clean up after analysis
-    rootfs.clean_up()
-    # we should now be able to set imported layers
+    # Doing analysis
+    print("Docker file given...")
+    dcom.default_common_procedure(
+        full_image, redo, driver, extension)
     lock.set_imported_layers(full_image)
-    # save to the cache
-    common.save_to_cache(full_image)
```

Comment:
`save_to_cache` happens here *after* `lock.set_imported_layers(full_image)` but with your change, it now happens before we set the imported layers. I think this might change what gets saved to the cache? Have you experimented with this?

Reply:
My bad, I overlooked it. Now I have fixed this. What I did is that I removed the `image_file` argument and used `args` as my parameter to check from where this function is called. This will also help me to reduce the arguments from 6 to 5.
If `args` == `Full_Image` -> Then locking will happen first
If `args` == `Base_Image` -> Then just pass
else -> report will be called.
Here is the code: 
```
def default_common_procedure(image, redo, driver, extension, args=None):
    """Setup the image object and anything else for analysis"""
    # Add a Notice object for each layer
    for layer in image.layers:
        origin_str = 'Layer {}'.format(layer.layer_index)
        layer.origins.add_notice_origin(origin_str)
    # Set up working directories and mount points
    rootfs.set_up()
    # analyze the image
    cimage.analyze(image, redo, driver, extension)
    if args=="Full_Image":
        lock.set_imported_layers(image)
    elif args=="Base_Image":
        pass
    else:
        report.report_out(args, image)
    # Add the image layers to the cache
    common.save_to_cache(image)
    # Clean up working directories and mount points
    rootfs.clean_up()
```